### PR TITLE
BC-1460 - Course menu actions

### PR DESCRIPTION
**Pull request by murat-merdoglu-hpi**
# Short Description
- Edit/delete, share, copy and invite features are added as a menu to the course title.

## Links to Ticket and related Pull-Requests
- https://ticketsystem.dbildungscloud.de/browse/BC-1460
- https://github.com/hpi-schul-cloud/schulcloud-server/pull/3394
- https://github.com/hpi-schul-cloud/dof_app_deploy/pull/214

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/pages/rooms/_id/index.unit.ts [hunk 1]
```diff
@@ -162,4 +162,274 @@ describe("@pages/rooms/_id/index.vue", () => {
...
+			).toBe(false);
+			expect(
+				findMenuItems(
+					wrapper.vm.$i18n.t("pages.room.courseTitleMenu.share"),
+					menuItems
+				)
+			).toBe(false);
+		});
+
+		it("should have 'Share Course' menu if 'FEATURE_COURSE_SHARE' flag set", () => {
```
Comment (MartinSchuhmacher) [inline src/pages/rooms/_id/index.unit.ts:218]: "...set to true"

File: src/pages/rooms/_id/index.unit.ts [hunk 2]
```diff
@@ -162,4 +162,274 @@ describe("@pages/rooms/_id/index.vue", () => {
...
+
+			expect(
+				findMenuItems(
+					wrapper.vm.$i18n.t("pages.room.courseTitleMenu.share"),
+					menuItems
+				)
+			).toBe(true);
+		});
+
+		it("should have 'Copy/Duplicate Course' menu if 'FEATURE_COURSE_COPY' flag set", () => {
```
Comment (MartinSchuhmacher) [inline src/pages/rooms/_id/index.unit.ts:232]: "...set to true"

File: src/pages/rooms/_id/index.unit.ts [hunk 3]
```diff
@@ -162,4 +162,274 @@ describe("@pages/rooms/_id/index.vue", () => {
...
+			const threeDotButton = wrapper.find(".three-dot-button");
+			await threeDotButton.trigger("click");
+			const moreActionButton = wrapper.find(`[data-testid=title-menu-invite]`);
+			await moreActionButton.trigger("click");
+
+			expect(createCourseInvitationSpy).toHaveBeenCalled();
+			expect(createCourseInvitationSpy.mock.calls[0][0]).toStrictEqual("123");
+		});
+
+		it("should call shareCourse method when 'Invite to the course' menu clicked", async () => {
```
Comment (MartinSchuhmacher) [inline src/pages/rooms/_id/index.unit.ts:299]: I guess the "Invite to the course" needs to be changed for shareCourse :)

**Review by MartinSchuhmacher — CHANGES_REQUESTED** [review]

File: src/store/room.ts [hunk 4]
```diff
@@ -45,6 +45,8 @@ export class Room extends VuexModule {
 		status: "",
 		message: "",
 	};
+	courseInvitationLink: string = "";
+	courseShareToken: string = "";
```
[the remark this answers is not in the document]
Comment (murat-merdoglu-hpi) [reply src/store/room.ts]: Changed.
Maybe we should work on access modifiers generally soon.

**Review by MartinSchuhmacher — APPROVED** [review]

**Pull request merged**